packages/spl-token/src/domain/main.ts

```diff
@@ -0,0 +1,147 @@
+import {
+  AccountIndexerRequestArgs,
+  IndexerMainDomain,
+  IndexerMainDomainContext,
+  IndexerMainDomainWithDiscovery,
+  IndexerMainDomainWithStats,
+} from '@aleph-indexer/framework'
+import { Token, solanaPrivateRPCRoundRobin } from '@aleph-indexer/core'
+import {
+  SPLAccountBalance,
+  SPLAccountHoldings,
+  SPLTokenEvent,
+  SPLTokenInfo,
+  SPLTokenType,
+} from '../types.js'
+import { discoveryFn } from '../utils/discovery.js'
+import { AccountHoldingsFilters, MintEventsFilters } from './types.js'
+import { TOKEN_PROGRAM_ID } from '../constants.js'
+
+export default class MainDomain
+  extends IndexerMainDomain
+  implements IndexerMainDomainWithDiscovery, IndexerMainDomainWithStats
+{
+  protected tokens: Record<string, SPLTokenInfo> = {}
+
+  constructor(protected context: IndexerMainDomainContext) {
+    super(context, {})
+  }
+
+  async updateStats(now: number): Promise<void> {
+    console.log('Method not implemented.')
+  }
+
+  async discoverAccounts(): Promise<AccountIndexerRequestArgs[]> {
+    const init = {
+      account: '',
+      index: {
+        transactions: {
+          chunkDelay: 0,
+          chunkTimeframe: 1000 * 60 * 60 * 24,
+        },
+        content: false,
+      },
+    }
+    return [init]
+  }
+
+  async init(...args: unknown[]): Promise<void> {
+    super.init(...args)
+    const { accounts, mints } = await discoveryFn()
+
+    await Promise.all(
+      accounts.map(async (account: string) => {
+        const connection = solanaPrivateRPCRoundRobin.getClient()
+        const mint = await Token.getTokenMintByAccount(
```

Comment:
I would move the mint discovery of an account to the discovery helper, doing some refactor on the types returned there